app/src/main/java/com/m2049r/xmrwallet/layout/TransactionInfoAdapter.java

```diff
@@ -73,6 +74,27 @@ public TransactionInfoAdapter(Context context, OnInteractionListener listener) {
         DATETIME_FORMATTER.setTimeZone(tz);
     }
 
+
+    private static class TransactionInfoDiff extends DiffCallback<TransactionInfo> {
+
+        public TransactionInfoDiff(List<TransactionInfo> oldList, List<TransactionInfo> newList) {
+            super(oldList, newList);
+        }
+
+        @Override
+        public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
+            return mOldList.get(oldItemPosition).hash.equals(
+                    mNewList.get(newItemPosition).hash
+            );
+        }
+
+        @Override
+        public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
+            return mOldList.get(oldItemPosition).isPending == mNewList.get(newItemPosition).isPending &&
```

Comment:
why compare only pending & failed status?

Reply:
Thanks m2049r I love your wallet!! I figured if pending has not changed and failure has not changed, it must be confirmed on chain, so the txinfo is final. I hope you can verify that logic because I think you will know this much better. These and others please see that they are compatible. I will be happy to update the PR for whatever changes are required.

Reply:
thanks :) this makes sense, i had also misread the `...hash.equals(...` in `areItemsTheSame()`.